### PR TITLE
Emote Cooldown  System

### PR DESCRIPTION
**Pull request by FalseIncarnate**
Introduces the Emote Cooldown System, designed to prevent spamming of
sound-based emotes.

Emotes that respect the cooldown:
- *beep
- *buzz
- *ping
- *squish
- *scream
- *fart
- *chirp
- *law

Emotes that are exempt:
- Everything not listed above

Emotes can be added to the system easily, and all cooldown-incurring
emotes share a single cooldown. This means you have to wait to *scream
after you *fart.

Also adds Tigercat's *squish emote ( #394 ), with sound credit to
DrMinky(Freesound.org). This emote is only usable by Slime People, and
accepts a parameter if you wish to squish at something/someone.

### CODE REVIEW

**Comment by Markolie** [comment]
*flip as well please.

**Comment by FalseIncarnate** [comment]
I won't be able to make any changes until later tonight. If Fox is willing, I'll copy the *flip code into my PR like I did with Tiger's *squish. I'll give *flip a cool down if it doesn't have an internal one so we can ensure it finishes it's animation before being usable again.

Sorry about the variable naming, I blame my past programming teachers. I'll fix that when I get home as well.

Also, *fart still has it's internal cool down in place, so you still can't successfully fart within one minute of your last one and must wait the cool down period between attempts. I wasn't sure of a better / balanced way handle it.

**Comment by FalseIncarnate** [comment]
Updated to be compatible with *flip, fixed monkeys not being able to emote, adjusted handle_emote_CD() to allow for manual disabling of cooldown emotes for a specific mob (you'll need +VAREDIT to do it though), and standardized the variable name as recommended by Mark.

**Comment by Fox-McCloud** [comment]
I suspect that if you don't do a merge branch master, you're going to wind up having flip in the codebase twice, once thsi gets merged in.

File: code/modules/mob/emote.dm [hunk 1]
```diff
@@ -1,3 +1,18 @@
+#define EMOTE_COOLDOWN 50
```
Comment (Fox-McCloud) [inline code/modules/mob/emote.dm]: being nitpicky here, but you might want to add in a comment that this is in deciseconds.

**Pull request closed without merging**